### PR TITLE
Automated canary builds for dev branch

### PR DESCRIPTION
**Pull request by nickwesselman**
## Description
When committing/merging to the dev branch, automatically perform a `lerna version` and `lerna publish` to push a new canary to npm.

## Motivation
Enable `jss create` to work against latest work on `dev` for internal QA and acceptance testing, as well as external developers who are curious or looking to contribute.

## How Has This Been Tested?
Commits to `dev-prerelease-publish` until success:
https://www.npmjs.com/package/@sitecore-jss/sitecore-jss/v/16.0.0-canary.6
(build steps will run against any branch prefixed `dev`)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

## Checklist:

- [x] I have read the Contributing guide.
- [x] My code follows the code style of this project.
- [x] My code/comments/docs fully adhere to the Code of Conduct.
- [ ] My change is a code change and it requires an update to the documentation.
- [ ] My change is a documentation change and it requires an update to the navigation.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -54,3 +57,20 @@ steps:
   env: {
     testEnv: 'ci'
   }
+- script: |
+    git config user.email builds@sitecore.com
+    git config user.name "Automated Build"
+    git remote set-url origin https://$GITHUB_PAT_ENV@github.com/Sitecore/jss.git
+    git checkout $(Build.SourceBranchName)
+    $(npm bin)/lerna version prerelease --preid canary --force-publish --message "version %s [skip ci]" --yes
```
Comment (ambrauer) [inline azure-pipelines.yml:65]: What's the significance of "[skip ci]" here?

**Review by ambrauer — CHANGES_REQUESTED** [review]

File: azure-pipelines.yml [hunk 2]
```diff
@@ -32,6 +32,9 @@ pool:
   # todo: we dgaf what specific version of Linux is installed, is there any way to wildcard the version?
   vmImage: 'Ubuntu-16.04'
 
+variables:
+  isDev: $[startsWith(variables['Build.SourceBranch'], 'refs/heads/dev')]
```
[the remark this answers is not in the document]
Comment (nickwesselman) [reply azure-pipelines.yml:36]: Yeah I planned to keep it there for future testing purposes but probably more dangerous than helpful. I'll change it.

File: azure-pipelines.yml (hunk 1 above)
Comment (nickwesselman) [reply azure-pipelines.yml:65]: Only documented in this [commit message](https://github.com/Sitecore/jss/pull/513/commits/b302e6756618ee6f204773afb445303c86ebd48b) but it [prevents Azure DevOps from running a build](https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/azure-repos-git?view=azure-devops&tabs=yaml#skipping-ci-for-individual-commits) for the commit (and thus triggering the version again).

I'll add a comment to the pipeline yml.

File: azure-pipelines.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nickwesselman) [reply azure-pipelines.yml]: Changed to `eq`

File: azure-pipelines.yml (hunk 1 above)
Comment (nickwesselman) [reply azure-pipelines.yml:75]: Added a bunch of comments :)

**Review by ambrauer — APPROVED** [review]

**Comment by ambrauer** [comment]
@nickwesselman Looks good to me, will wait for @anastasiya29 and @sc-illiakovalenko to have a look.

**Review by sc-illiakovalenko — APPROVED** [review]
Looks good!

**Pull request merged**